### ISSUE TITLE
Should we establish a release schedule?

### ISSUE THREAD

**Issue by PuercoPop** [stumpwm/stumpwm#454]
Over at [`#stumpwm`](https://irclog.tymoon.eu/freenode/stumpwm?from=1524643524) we were asked to make a new release for the purpose of using the tarball for OpenBSD ports. How about tagging master every 6 months as a release? Maybe using YYYY-{1,2} instead of a version number.

**Comment by dbjergaard** [comment]
Deal! I've been considering this for a while.  I think this model will fit us better since we're not adding new functionality that could mark major releases.

I was thinking "june" but that puts the next release just around Christmas. as much as this would delight our users, I wouldn't be able to consistently get it out.  Instead I propose May and November.  I also think we should do YY.MM for the version number.  That means the next version will be 18.05 and we'll release it next month sometime.

**Comment by PuercoPop** [comment]
@dbjergaard Awesome! Good call on not scheduling a release on December; plus may is just around the corner!

**Comment by Harleqin** [comment]
I have a few opinions about this.

The year 18 was 2000 years ago.  I am old enough to remember the Y2K mess, so I am firmly opposed to shortening year representations.  Even if “it doesn't matter”, I don't want to have to think about where it matters or not.

The other thing I don't like is having things that look like decimals, but aren't, and contain leading zeroes after the dot.  Is 18.05 0.45 less than 18.5?

If the releases are to be named after dates, I'd use something close to ISO 8601, so e. g. `2018-05`.

Does making a difference between _version_ and _release_ make sense, so that e. g. version 1.1.0 is released as release 2018-05?

**Comment by Harleqin** [comment]
One more thought: maybe the release name should be exactly the date, e. g. 2018-05-15.

**Comment by dbjergaard** [comment]
I appreciate your opinions, however I still stand by the YY.MM version numbers. Its succinct, and loosely coupled to the year.  It can be ordered, and if you know the version number, you know roughly the date.  They aren't supposed to be super tightly coupled, in 2100, the version number will be 100.05. No one interprets the standard X.Y.Z version numbers as decimals either. Finally, this versioning system has precedent in other open source projects (specifically a very popular linux distro).

**Comment by Harleqin** [comment]
Of course, I can't tell you how to name the releases, I can only give recommendations.

A few more thoughts:

- Loose coupling becomes hard coupling as soon as someone depends on it programmatically.

- If it is not decimal, why not 18.5 and 18.11?

- Having precedent seems not to be a universally endearing feature to me, given the current state of the art.

**Comment by jbaber** [comment]
- *A* reason for leading zeros is to make things sort chronologically by default
```bash
$ ls /tmp/good |sort
18.11
18.5
19.11
19.5
20.11
20.5
21.11
21.5
$ ls /tmp/great |sort
18.05
18.11
19.05
19.11
20.05
20.11
21.05
21.11
$
```

Not being decimal numbers definitively frees us to use leading zeros.

**Comment by Jehops** [comment]
I was [altered](https://portscout.freebsd.org/jrm@freebsd.org.html) to a new release assuming that this was the May release.  But, this was just a renaming of the tag, right?

**Comment by dbjergaard** [comment]
It was, I will try to release 18.05 in the next few days.

**Comment by PuercoPop** [comment]
@dbjergaard Reading the [release notes](https://github.com/stumpwm/stumpwm/releases/tag/18.05) I think  

> Dropped support for SBCL.

should prolly read

> Only support SBCL

**Comment by Jehops** [comment]
> I've released 18.05 as a release candidate.

Thank you!

> If no issues come up in the next few days I'll remove the release-candidate tag.

I assume you are referring to the checkbox used to label the release as a "Pre-release" and not the tag?  If not, would you consider making releases immutable?  When a release changes (tag or tarball), it breaks our (FreeBSD) port/package.  I assume this would also be a problem for other packages.

**Comment by scifisamurai** [comment]
Considering sbcl is still mentioned in the README I really do hope that the release notes saying that sbcl is being removed is a typo. 

If it's not a typo & sbcl is being removed could someone explain why?  Thanks.

**Comment by Jehops** [comment]
It's clearly just a mistake.  AFAIR, only SBCL is supported  now.

**Comment by dbjergaard** [comment]
It’s a typo. I didn’t realize editing releases messed up package managers. I’ll make a new release and rename this one. In the future I’ll mark release candidates with an rc.
